sycl/source/detail/event_impl.cpp

```diff
@@ -126,6 +128,10 @@ void event_impl::setContextImpl(const ContextImplPtr &Context) {
   MIsContextInitialized = true;
 }
 
+event_impl::event_impl(bool)
+    : MIsInitialized(false), MHostEvent(false), MIsFlushed(true),
+      MState(HES_Complete) {}
```

Comment:
Could this be combined with the `event_impl(HostEventState State = HES_Complete)` constructor into something like `event_impl(HostEventState State = HES_Complete, bool IsHostEvent = true)`?

Reply:
No. We need a new constructor for event_impl that'll create non-host events that the argumentless event() constructor can call.   Meanwhile, we leave in place the (mostly host) temporary event_impl that the scheduler constructs.

Reply:
This was discussed offline and an alternative was found.